clustered/server/src/main/java/org/ehcache/clustered/server/EhcachePassiveEntity.java

```diff
@@ -0,0 +1,172 @@
+import org.ehcache.clustered.common.internal.messages.LifecycleMessage;
+import org.ehcache.clustered.common.internal.messages.LifecycleMessage.ConfigureStoreManager;
+import org.ehcache.clustered.common.internal.messages.LifecycleMessage.CreateServerStore;
+import org.ehcache.clustered.common.internal.messages.LifecycleMessage.DestroyServerStore;
+import org.ehcache.clustered.server.state.EhcacheStateService;
+import org.ehcache.clustered.server.state.config.EhcacheStateServiceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terracotta.entity.BasicServiceConfiguration;
+import org.terracotta.entity.PassiveServerEntity;
+import org.terracotta.entity.ServiceRegistry;
+import org.terracotta.offheapresource.OffHeapResources;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.UUID;
+
+class EhcachePassiveEntity implements PassiveServerEntity<EhcacheEntityMessage, EhcacheEntityResponse> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(EhcachePassiveEntity.class);
+
+  private final UUID identity;
+  private final Set<String> offHeapResourceIdentifiers;
+  private final EhcacheStateService ehcacheStateService;
+
+  private final ServerStoreCompatibility storeCompatibility = new ServerStoreCompatibility();
+
+  EhcachePassiveEntity(ServiceRegistry services, byte[] config) {
+    this.identity = ClusteredEhcacheIdentity.deserialize(config);
+    OffHeapResources offHeapResources = services.getService(new BasicServiceConfiguration<OffHeapResources>(OffHeapResources.class));
+    if (offHeapResources == null) {
+      this.offHeapResourceIdentifiers = Collections.emptySet();
+    } else {
+      this.offHeapResourceIdentifiers = offHeapResources.getAllIdentifiers();
+    }
+    ehcacheStateService = services.getService(new EhcacheStateServiceConfig(services, this.offHeapResourceIdentifiers));
+    if (ehcacheStateService == null) {
+      throw new AssertionError("Server failed to retrieve EhcacheStateService.");
+    }
+  }
+
+  @Override
+  public void invoke(EhcacheEntityMessage message) {
+
+    try {
+      if (this.offHeapResourceIdentifiers.isEmpty()) {
+        throw new ServerMisconfigurationException("Server started without any offheap resources defined." +
+                                                  " Check your server configuration and define at least one offheap resource.");
+      }
+
+      switch (message.getType()) {
+        case LIFECYCLE_OP:
+          invokeLifeCycleOperation((LifecycleMessage) message);
+          break;
+        case SERVER_STORE_OP:
+          //TODO : #1211
+          break;
+        default:
+          throw new IllegalMessageException("Unknown message : " + message);
+      }
+    } catch (Exception e) {
+      LOGGER.error("Unexpected exception raised during operation: " + message, e);
+    }
+
+  }
+
+  private void invokeLifeCycleOperation(LifecycleMessage message) throws ClusterException{
```

Comment:
Forget what I said ... unless we hide the difference in parameters between a message with or without a `ClientDescriptor`